majilove/models.py

```diff
@@ -696,3 +708,102 @@ def __unicode__(self):
+
+    ATTRIBUTE_CHOICES = (
+        ('star', _('Star')),
+        ('shine', _('Shine')),
+        ('dream', _('Dream')),
+    )
+
+    i_attribute = models.PositiveIntegerField(_('Attribute'), choices=i_choices(ATTRIBUTE_CHOICES), null=True)
+
+    '''
+    ABOUT UNIT_CHOICES & SUBUNIT_CHOICES
+    - instead of unit/subunit choices, I think we want singer choices, but there are many combinations...
+    - link to Idol models?
+
+    ABOUT VERSIONS_CHOICES
+    - not sure how game version impacts song choices (I don't think it does)
+
+    ABOUT LOCATIONS_CHOICES
+    - we probably don't need this, SL songs are created equal
+    - unless this is where we care about Normal vs Campaign songs?
+
+    unlock: (Normal Songs only)
+    - Normal songs unlock by playing parts of the main story 
+    - I forget what these messages look like
+
+    daily: --> Campaign Songs
+
+    '''
+
+    release = models.DateTimeField(_('Release date'), null=True)
+    #itunes_id = models.PositiveIntegerField(_('Preview'), help_text='iTunes ID', null=True)
+    length = models.PositiveIntegerField(_('Length'), help_text=_('in seconds'), null=True)
+    #bpm = models.PositiveIntegerField(_('Beats per minute'), null=True)
+
+    @property
+    def length_in_minutes(self):
+        return time.strftime('%M:%S', time.gmtime(self.length))
+
+    SONGWRITERS = (
+        ('lyricist', _('Lyricist')),
+        ('composer', _('Composer')),
+        ('arranger', _('Arranger')),
+    )
+
+    lyricist = models.CharField(_('Lyricist'), max_length=100, null=True)
+    composer = models.CharField(_('Composer'), max_length=100, null=True)
+    arranger = models.CharField(_('Arranger'), max_length=100, null=True)
+
+    
+    easy_notes = models.PositiveIntegerField(string_concat('EASY', ' - ', _('Notes')), null=True)
+    easy_difficulty = models.PositiveIntegerField(string_concat('EASY', ' - ', _('Difficulty')), validators=DIFFICULTY_VALIDATORS, null=True)
+    normal_notes = models.PositiveIntegerField(string_concat('NORMAL', ' - ', _('Notes')), null=True)
+    normal_difficulty = models.PositiveIntegerField(string_concat('NORMAL', ' - ', _('Difficulty')), validators=DIFFICULTY_VALIDATORS, null=True)
+    hard_notes = models.PositiveIntegerField(string_concat('HARD', ' - ', _('Notes')), null=True)
+    hard_difficulty = models.PositiveIntegerField(string_concat('HARD', ' - ', _('Difficulty')), validators=DIFFICULTY_VALIDATORS, null=True)
+    pro_notes = models.PositiveIntegerField(string_concat('PRO', ' - ', _('Notes')), null=True)
+    pro_difficulty = models.PositiveIntegerField(string_concat('PRO', ' - ', _('Difficulty')), validators=DIFFICULTY_VALIDATORS, null=True)
+    master_notes = models.PositiveIntegerField(string_concat('MASTER', ' - ', _('Notes')), null=True)
+    master_difficulty = models.PositiveIntegerField(string_concat('MASTER', ' - ', _('Difficulty')), validators=DIFFICULTY_VALIDATORS, null=True)
+
```

Comment:
Song model (WIP). Still need to implement associated idols & song type (normal/campaign/event).